src/theory/arith/nl/icp/icp_solver.cpp

```diff
@@ -350,7 +350,8 @@ void ICPSolver::check()
     std::vector<Node> lemmas = generateLemmas();
     for (const auto& l : lemmas)
     {
-      d_im.addPendingArithLemma(l, InferenceId::NL_ICP_PROPAGATION);
+      d_im.addPendingArithLemma(Rewriter::rewrite(l),
```

Comment:
Does it make sense to always rewrite within addPendingArithLemma instead of making a special call here?

Reply:
I just removed it. I think we want to (always) rewrite within `addPendingArithLemma` to improve caching in the arithmetic inference manager.